src/App.tsx

```diff
@@ -1,56 +1,90 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
+
+const debounce = (
+  f: React.Dispatch<React.SetStateAction<string>>,
+  delay: number,
+) => {
+  let timerId: number;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [personName, setPersonName] = useState<string | null>(null);
+  const [dropdownRegulator, setDropdownRegulator] = useState(false);
```

Comment:
Use more descriptive name
For example: 
```suggestion
  const [isDropdownActive, setIsDropdownActive] = useState(false);
```